example/basic/basic_main.dart

```diff
@@ -4,17 +4,23 @@ import 'package:jaguar_serializer/serializer.dart';
 
 part 'basic_main.g.dart';
 
-@GenSerializer()
+@GenSerializer(addTypeInfo: true)
```

Comment:
with you proposition, after generation, one have no choice to embeded or not the typeInfo, that is why I think it is better to keep that option at runtime, `Serializer.encode(object, withTypeInfo: true);`